blocks.js

```diff
@@ -35,7 +35,10 @@ module.exports = function (file, block_size, cache) {
         var cb = cbs[i]
         cbs[i] = null
         if(!err) cache.set(i, buf)
-        while(cb.length) cb.shift()(err, buf, bytes_read)
+        var l = cb.length
+        for (var j = 0; j < cb.length; ++j)
+          cb[j](err, buf, bytes_read)
+        cb = cb.slice(l)
```

Comment:
this line leave `cb` in the same state but I'm not sure it's needed ... `cb` was created in this function scope and doesn't get any further use?

Reply:
Right, cbs[i] is nulled. Good catch. Thanks!